src/test/scala/nl/knaw/dans/easy/stage/EasyStageDatasetSpec.scala

```diff
@@ -16,156 +16,144 @@
+class EasyStageDatasetSpec extends FlatSpec with Matchers with OneInstancePerTest {
+  private val testDir = Paths.get("target/test", getClass.getSimpleName)
+  FileUtils.deleteQuietly(testDir.toFile)
+  Files.createDirectories(testDir)
 
-  private val tmpProps = new File("cfg/application.properties")
+  private val testBagMedium = Paths.get("src/test/resources/dataset-bags/medium")
 
   "createFileAndFolderSdos" should "stumble if the data folder does not exist" in {
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data") // Note that this directory is never created, on purpose.
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(bagDir)
 
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    bagitDir.mkdirs()
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Failure[_]]
-
-    deleteDirectory(bagitDir)
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Failure[_]]
   }
 
   it should "be happy with an empty data folder" in {
-
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    dataDir.mkdirs()
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
-    sdoSetDir.exists() shouldBe false
-
-    deleteDirectory(bagitDir)
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data")
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(dataDir)
+
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
+    Files.exists(sdoSetDir) shouldBe false
   }
 
   it should "stumble over a manifest-sha1.txt with too many fields on a line" in {
-
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    dataDir.mkdirs()
-    FileUtils.write(new File(bagitDir,"manifest-sha1.txt"),"a b c")
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS).failed.get should
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data")
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(dataDir)
+    FileUtils.write(bagDir.resolve("manifest-sha1.txt").toFile,"a b c")
+
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS).failed.get should
```

Comment:
I agree that it looks a bit clumsy and probably doesn't generate a very nice error message if the result happens not to be a `Failure`. This line was already like this, by the way.

By the way, I prefer to include separate checks like this:
```scala
val result = functionUnderTestReturningTry
result shouldBe a[Failure[_]]
inside(result) {
  case Failure(e) => e should have message "Whatever"
}
```
Then, if the result is *not* a `Failure` the error report will be clearer.